.github/workflows/tests.yml

```diff
@@ -52,7 +52,17 @@ jobs:
       # Build the docs
       - name: Build docs to store
         run: |
-          sphinx-build -b html docs/ docs/_build/html -W --keep-going
+          sphinx-build -b html docs/ docs/_build/html --keep-going -w warnings.txt
+
+      - name: Check that there are no unexpected Sphinx warnings
+        if: matrix.python-version == '3.9'
+        shell: python
+        run: |
+          from pathlib import Path
+          text = Path("./warnings.txt").read_text().strip()
+          print("\n=== Sphinx Warnings ===\n\n" + text)  # Print just for reference so we can look at the logs
+          unexpected = [ii for ii in text.split("\n") if "kitchen-sink" not in ii]
+          assert len(unexpected) == 0
```

Comment:
The reason for this manual check instead of `-W` is to not error on warnings in the kitchen sink pages? (since those are out of our control?)

Reply:
The manual check is because we **know** the kitchen sink has errors, and that is because we want to be able to demo what it looks like when an error is rendered into the actual page, since many of our users will do this. So we want to make sure that there are no "unexpected errors", which is what this check is for

Reply:
OK, great!